game.py

```diff
@@ -17,25 +17,47 @@ def snowman(snowman_word):
     """Complete the snowman function
     replace "pass" below with your own code
     It should print 'Congratulations, you win!'
-    If the player wins and, 
+    If the player wins and,
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    wrong_guesses_list = []
+
+    while (not is_word_guessed(snowman_word, correct_letter_guess_statuses)
+           and len(wrong_guesses_list) != SNOWMAN_MAX_WRONG_GUESSES):
+        user_guess = get_letter_from_user(correct_letter_guess_statuses,
+                                          wrong_guesses_list)
+
+        if user_guess in correct_letter_guess_statuses:
+            print("You guessed a letter that's in the word!")
+            correct_letter_guess_statuses[user_guess] = True
+        else:
+            print(f"The letter {user_guess} is not in the word")
+            wrong_guesses_list.append(user_guess)
+
+        print_snowman_graphic(len(wrong_guesses_list))
+        print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+        print(f"Wrong guesses: {wrong_guesses_list}")
```

Comment:
Overall this looks good, but when everything does get printed out in the console, it does get to be a little cluttered. When working with console output, it can sometimes be a good idea to add in some line breaks just to make the flow of the game a little more readable and easy to follow!